src/poker.js

```diff
@@ -1,10 +1,69 @@
-function winningPair() {}
+const cardNumbers = {
+  J: '11',
+  Q: '12',
+  K: '13',
+  A: '14'
+}
+
+function pairCheck(pair) {
+  return pair[0] === pair[1]
+}
+
+function setCheck(hand) {
+  return hand.length === 3 && hand[0] === hand[1] && hand[1] === hand[2]
+}
+
+const checkNumber = (pairs) =>
+  pairs.map((pair) => Number(cardNumbers[pair] || pair))
+
+function winningPair(pair1, pair2) {
+  if (!pairCheck(pair1) && !pairCheck(pair2)) return []
+  else if (pairCheck(pair1) && pairCheck(pair2)) {
+    const pair1Num = checkNumber(pair1)
+    const pair2Num = checkNumber(pair2)
+    return pair1Num[0] > pair2Num[0] ? pair1 : pair2
+  } else if (!pairCheck(pair1)) return pair2
+  else return pair1
+}
 
 // Extension criteria
 
-function winningPairFromArray() {}
+function winningPairFromArray(pairs) {
+  pairs = pairs.filter(pairCheck)
 
-function winning3CardHand() {}
+  let bestPair
+
+  for (const pair of pairs) {
+    if (bestPair == null) {
+      bestPair = pair
+      continue
+    }
+
+    if (checkNumber(pair)[0] > checkNumber(bestPair)[0]) {
+      bestPair = pair
+    }
+  }
+  return bestPair ?? []
+}
+
+function winning3CardHand(hands) {
+  const setHands = hands.filter(setCheck)
+  if (setHands.length === 0) return winningPairFromArray(hands)
```

Comment:
You can simplify this in js by doing `if (!setHands.length) {}` since 0 is a falsy value